front_matter/create_extractor.ts

```diff
@@ -1,11 +1,12 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 
 import {
-  Format,
   MAP_FORMAT_TO_EXTRACTOR_RX,
   MAP_FORMAT_TO_RECOGNIZER_RX,
 } from "./_formats.ts";
 
+type Format = "yaml" | "toml" | "json" | "unknown";
```

Comment:
Should this be exported?

Reply:
Nice catch, I would think so.

Reply:
The name 'Format' is currently exported as enum value, which this PR deprecates. I think exporting this 'Format' in this PR could break typing of the user's code.

How about using literal type (`"yaml" | "toml" | "json" | "unknown"`) directly in public interfaces?